pkg/client/dtclient/config_client.go

```diff
@@ -573,6 +540,62 @@ func (d *DynatraceClient) getExistingValuesFromEndpoint(ctx context.Context, the
 	return existingValues, nil
 }
 
+func (d *DynatraceClient) findUniqueByName(ctx context.Context, theApi api.API, values []Value, objectName string) (string, error) {
+	var objectId = ""
+	var matchingObjectsFound = 0
+	for i := 0; i < len(values); i++ {
+		value := values[i]
+		if value.Name == objectName || escapeApiValueName(ctx, value) == objectName {
+			if matchingObjectsFound == 0 {
+				objectId = value.Id
+			}
+			matchingObjectsFound++
+		}
+	}
+
+	if matchingObjectsFound > 1 {
+		log.WithCtxFields(ctx).Warn("Found %d configs with same name: %s. Please delete duplicates.", matchingObjectsFound, objectName)
+	}
+
+	if objectId != "" {
+		log.WithCtxFields(ctx).Debug("Found existing config %s (%s) with id %s", objectName, theApi.ID, objectId)
+	}
+
+	return objectId, nil
+}
+
+func escapeApiValueName(ctx context.Context, value Value) string {
+	valueName, err := template.EscapeSpecialCharactersInValue(value.Name, template.FullStringEscapeFunction)
+	if err != nil {
+		log.WithCtxFields(ctx).Warn("failed to string escape API value '%s' while checking if object exists, check directly", value.Name)
+		return value.Name
+	}
+	return valueName.(string)
+}
+
+func (d *DynatraceClient) findUnique(values []Value, payload []byte, checkEqualFunc func(map[string]any, map[string]any) bool) (string, error) {
+	if checkEqualFunc == nil {
+		return "", nil
+	}
+	var objectId = ""
+	var matchingObjectsFound = 0
+	for i := 0; i < len(values); i++ {
+		value := values[i]
+		var pl map[string]any
+		if err := json.Unmarshal(payload, &pl); err != nil {
+			return "", err
+		}
+		if checkEqualFunc(value.CustomFields, pl) {
+			if matchingObjectsFound == 0 {
+				objectId = value.Id
+			}
+			matchingObjectsFound++
+		}
+	}
+
+	return objectId, nil
```

Comment:
you're now missing the warning and debug logs that the findUniqueByName has on lines 556-562. I think it's worth having that information in the logs for custom equals functions too